src/reducers/featureReducer.js

```diff
@@ -0,0 +1,66 @@
+import {
+  SET_NEXT_SLIDE,
+  SET_PREVIOUS_SLIDE,
+  SET_INIT_SLIDE,
+} from '../actions/types';
+
+const initialState = {
+  artList: [],
+  activeList: [],
+};
+
+/**
+ *  @param {function} state
+ * @param {function} action
+ *  @returns {Object} action obj
+ */
+function featureReducer(state = initialState, action) {
+  if (action.type === SET_INIT_SLIDE) {
+    return Object.assign({}, state, {
```

Comment:
Why do you prefer this to using the spread operator.

Reply:
I needed a way not to mutate the state, I searched MDN and I found it. However, the spread syntax is a Stage 4 proposal for ECMAScript and accepted for the 2018 specification release. Nevertheless will adopt it.